pkg/sethandler/controller.go

```diff
@@ -158,9 +164,11 @@ func determineContainersToBeSet(oldPod, newPod v1.Pod) map[string]int {
 func gatherAllContainers(pod v1.Pod) map[string]int {
 	workingContainers := map[string]int{}
 	for _, containerStatus := range pod.Status.ContainerStatuses {
-		if containerStatus.ContainerID == "" {
+		// Ignore containers that do not have a container id or not running
+		if containerStatus.ContainerID == "" || containerStatus.State.Running == nil {
```

Comment:
we actually did have "Running" check in the past, but we have seen strange things happening in production which prompted us to not rely on this status indication. this is when we introduced adding an annotation instead to signal the Pod was already adjusted

you can find some background here: https://github.com/nokia/CPU-Pooler/pull/43#issuecomment-635338842

Reply:
looking back at the legacy code tho we were checking for Pod status Running and Container status Ready, while I see you are checking fo Container status Running here

how reliable is this status, can you elaborate when it is set exactly?

Reply:
Ah, sorry I had missed that this was previously removed! I'll investigate further to see if this is reliable/find a reliable method.

I'm happy if you want to decline this PR for now as it might be a while for me to get to the bottom of the issue

Reply:
the first two parts of the PR are definitely useful, and welcomed on their own!

sorry for the late reply but we are also in the process of figuring out a more reliably, and fool-proof way of ensuring every container gets their CPUSet reliably provisioned in a reasonable manner.
unfortunetaly this state check would not work in some scenarios. we are drafting A PR to address this via changin the architecture to be multi-threaded, and include some retry mechanism to work around timing issues instead

so please remove this part, and stay tuned for our follow up :) but the new update mechanism and the Job state check would be absolutely welcomed!